libsolutil/CommonData.cpp

```diff
@@ -188,5 +188,32 @@ string solidity::util::formatAsStringOrNumber(string const& _value)
 		if (c <= 0x1f || c >= 0x7f || c == '"')
 			return "0x" + h256(_value, h256::AlignLeft).hex();
 
-	return "\"" + _value + "\"";
+	return escapeAndQuoteString(_value);
+}
+
+
+string solidity::util::escapeAndQuoteString(string const& _input)
+{
+	std::string output = "\"" + _input + "\"";
+
+	for (string::iterator it = output.begin(); it != output.end(); it++)
+		if (
+			(*it == '"' && it != output.begin() && it != output.end() - 1)
```

Comment:
I added the `"`s already at the beginning (to avoid an extra allocation at the end) thus I am excluding them from the check here

Reply:
Can we keep it modular, please?

Reply:
You mean don't do the quoting inside this function?

Reply:
If there is a way to do it without additional comments and a way to do it with additional comments, please always choose the former.